compiler/lib/flow.ml

```diff
@@ -337,6 +337,21 @@ let the_def_of info x =
         x
   | Pc c -> Some (Constant c)
 
+(* If [constant_identical a b = Some true], then the two values cannot be
+   distinguished, i.e., they are not different objects (and [caml_js_equals a b
+   = true]) and if both are floats, they are bitwise equal. *)
+let constant_identical a b =
```

Comment:
This function could just return a boolean (meaning that we know that the two constants cannot be distinguished).